contrib/pipelines/core/pipeline.go

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package core
+
+import (
+	"encoding/json"
+
+	"github.com/skydive-project/skydive/flow"
+	"github.com/skydive-project/skydive/logging"
+	ws "github.com/skydive-project/skydive/websocket"
+)
+
+// CfgRoot configuration root path
+const CfgRoot = "pipeline."
+
+// Pipeline manager
+type Pipeline struct {
+	Transformer Transformer
+	Classifier  Classifier
+	Filterer    Filterer
+	Encoder     Encoder
+	Compressor  Compressor
+	Storer      Storer
+}
+
+// NewPipeline defines the pipeline elements
+func NewPipeline(transformer Transformer, classifier Classifier, filterer Filterer, encoder Encoder, compressor Compressor, storer Storer) *Pipeline {
```

Comment:
How complicated it will be to request that this will be configurable so that it will be possible to have just subset of those in a pipeline and does it make sense to also require 
(1) Change in the order of things ... so you can have classifier before transformer as an example
(2) Allow more than one step so for example transformer1 + transformer2 + ... 

This might require more work and be second phase but I want to understand if this is possible and if this make sense ?

Reply:
Great comments ...

(1) so changing the order of steps - I am not so sure:
- for example subscribing to events always must come before storing flows to target
- for example (2) encoding always comes before compression
But if you can think of specific cases when this makes sense then we can consider

(2) I generally agree - we may want to setup a list of tranformers - each using the previous output as its' input - we can implement this once we have a first concrete use case